### DISCUSSION TITLE
Focus lost when using AlpineJS $id after the Livewire request is sent

### DISCUSSION THREAD

**Discussion by johanvanhelden** [livewire/livewire#4967] [7. Bugs (archived)]
Since it's still present right now in the most recent version of Livewire/Alpine I've decided to turn it into a bug. As more people are experiencing it.

I hope someone can help me out how to fix this.

### Discussed in https://github.com/livewire/livewire/discussions/4790

<div type='discussions-op-text'>

<sup>Originally posted by **johanvanhelden** March 10, 2022</sup>
Hi all! Not sure if this is a bug, or me not understanding something.

I was watching Caleb's Laracon talk and he was showing of how to use Alpine to generate ids for elements.

So I just finished implementing it, but after I type some data into an input element, and the Livewire request is sent to the server, the input looses its focus.

**Demo:**
![LivewireAlpineDemo](https://user-images.githubusercontent.com/19389981/157751947-bbf80441-2eb7-4e46-a9d1-15b1fa32fa1c.gif)

The HTML for the input element looks like this:
```html
<div x-data="" x-id="['input']">
    <label class="block | text-sm font-medium text-gray-700 mb-2" x-bind:for="$id('input')" for="input-2">Name</label>
    <div>
        <input 
            x-bind:id="$id('input')" 
            type="text" 
            class="w-full block | shadow-sm rounded border-gray-300 sm:text-sm | focus:ring-blue-500 focus:border-blue-500" 
            wire:model="state.name" 
            autocomplete="name" 
            autofocus="autofocus" 
            id="input-2"
        >
    </div>
</div>
```

If I revert the `$id()` changes, everything works like it should, without anything loosing focus.</div>

**Comment by joshhanley** [comment]
@johanvanhelden please don't open a seperate discussion, just change the category on the original discussion to bug 🙂
